test/testffmpeg_vulkan.c

```diff
@@ -0,0 +1,973 @@
+        RESULT_CASE(VK_ERROR_EXTENSION_NOT_PRESENT);
+        RESULT_CASE(VK_ERROR_FEATURE_NOT_PRESENT);
+        RESULT_CASE(VK_ERROR_INCOMPATIBLE_DRIVER);
+        RESULT_CASE(VK_ERROR_TOO_MANY_OBJECTS);
+        RESULT_CASE(VK_ERROR_FORMAT_NOT_SUPPORTED);
+        RESULT_CASE(VK_ERROR_FRAGMENTED_POOL);
+        RESULT_CASE(VK_ERROR_SURFACE_LOST_KHR);
+        RESULT_CASE(VK_ERROR_NATIVE_WINDOW_IN_USE_KHR);
+        RESULT_CASE(VK_SUBOPTIMAL_KHR);
+        RESULT_CASE(VK_ERROR_OUT_OF_DATE_KHR);
+        RESULT_CASE(VK_ERROR_INCOMPATIBLE_DISPLAY_KHR);
+        RESULT_CASE(VK_ERROR_VALIDATION_FAILED_EXT);
+        RESULT_CASE(VK_ERROR_OUT_OF_POOL_MEMORY_KHR);
+        RESULT_CASE(VK_ERROR_INVALID_SHADER_NV);
+#undef RESULT_CASE
+    default:
+        break;
+    }
+    return (result < 0) ? "VK_ERROR_<Unknown>" : "VK_<Unknown>";
+}
+
+static int createInstance(VulkanVideoContext *context)
+{
+    static const char *optional_extensions[] = {
+        VK_EXT_SWAPCHAIN_COLOR_SPACE_EXTENSION_NAME,
+        VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME
+    };
+    VkApplicationInfo appInfo = { 0 };
+    VkInstanceCreateInfo instanceCreateInfo = { 0 };
+    VkResult result;
+    char const *const *instanceExtensions = SDL_Vulkan_GetInstanceExtensions(&instanceCreateInfo.enabledExtensionCount);
+
+    appInfo.sType = VK_STRUCTURE_TYPE_APPLICATION_INFO;
+    appInfo.apiVersion = VK_API_VERSION_1_3;
+    instanceCreateInfo.sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO;
+    instanceCreateInfo.pApplicationInfo = &appInfo;
+
+    const char **instanceExtensionsCopy = SDL_calloc(instanceCreateInfo.enabledExtensionCount + SDL_arraysize(optional_extensions), sizeof(const char *));
+    for (uint32_t i = 0; i < instanceCreateInfo.enabledExtensionCount; i++) {
+        instanceExtensionsCopy[i] = instanceExtensions[i];
+    }
+
+    // Get the rest of the optional extensions
+    {
+        uint32_t extensionCount;
+        if (context->vkEnumerateInstanceExtensionProperties(NULL, &extensionCount, NULL) == VK_SUCCESS && extensionCount > 0) {
+            VkExtensionProperties *extensionProperties = SDL_calloc(sizeof(VkExtensionProperties), extensionCount);
+            if (context->vkEnumerateInstanceExtensionProperties(NULL, &extensionCount, extensionProperties) == VK_SUCCESS) {
+                for (uint32_t i = 0; i < SDL_arraysize(optional_extensions); ++i) {
+                    for (uint32_t j = 0; j < extensionCount; ++j) {
+                        if (SDL_strcmp(extensionProperties[j].extensionName, optional_extensions[i]) == 0) {
+                            instanceExtensionsCopy[instanceCreateInfo.enabledExtensionCount++] = optional_extensions[i];
+                            break;
+                        }
+                    }
+                }
+            }
+            SDL_free(extensionProperties);
+        }
+    }
+    instanceCreateInfo.ppEnabledExtensionNames = instanceExtensionsCopy;
```

Comment:
I don't see where you free instanceExtensionsCopy

Reply:
It's at the end, but after a potential error return. I'll move that as a separate commit.

Reply:
Whoops, I was talking about SDL_render_vulkan.c. I'll fix that here, thanks!